### PR TITLE
Convert hashicorp/terraform-helm to GitHub Actions

### PR DESCRIPTION
**Pull request by hc-github-team-es-release-engineering**
Pipeline migrated from [CircleCI](https://app.circleci.com/pipelines/github/hashicorp/terraform-helm) :tada:

## Manual steps

Perform the following steps to complete the migration:

### hashicorp/terraform-helm/test
- [x] Ensure environment variable is updated: `SLACK_WEBHOOK`

### hashicorp/terraform-helm/update-helm-charts-index
- [x] Ensure environment variable is updated: `SLACK_WEBHOOK`

### CODE REVIEW

File: .github/workflows/update-helm-charts-index.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            -f SOURCE_TAG="${{ github.ref_name }}" \
+            -f SOURCE_REPO="${{ github.repository }}"
+
+      - name: send slack status
+        if: failure()
+        uses: hashicorp/actions-slack-status@v1
+        with:
+          status: "failure"
+          slack_webhook: ${{ secrets.SLACK_WEBHOOK }}
+          failure_message: "Failed to trigger an update to the helm charts index."
```
Comment (dekimsey) [inline .github/workflows/update-helm-charts-index.yml:45]: I think these are `slack-webhook-url` and `failure-message` respectively.

**Review by dekimsey — APPROVED** [review]
Minor correction, but it all looks good

**Review by dekimsey — APPROVED** [review]

**Review by arybolovlev — APPROVED** [review]
Thank you for taking care of this!

**Comment by sarahethompson** [comment]
I have added the HELM_CHARTS_GITHUB_TOKEN and SLACK_WEBHOOK secrets to the repo. See https://hashicorp.slack.com/archives/C03UVR4C944/p1681984578646699 for more info. Going to merge this in now.

**Comment by sarahethompson** [comment]
@dekimsey @smacfarlane just double checking, but do we also want to keep the unit test job? https://github.com/hashicorp/terraform-helm/blob/bd954ab46c414dea3d567a8b9dcf0f6d1b77d286/.circleci/config.yml#L6

**Pull request merged**